Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -272,10 +274,14 @@ private func createResolvedPackages(
         for targetBuilder in packageBuilder.targets {
             // If a target with similar name was encountered before, we emit a diagnostic.
             let targetName = targetBuilder.target.name
-            if allTargetNames.contains(targetName) {
-                diagnostics.emit(ModuleError.duplicateModule(targetName), location: diagnosticLocation())
```

Comment:
I think we should start a search for the offending packages when we hit this to avoid making this loop more complicated.